Sources/StreamChat/Database/DataStore.swift

```diff
@@ -80,4 +80,17 @@ public struct DataStore {
     public func message(id: MessageId) -> ChatMessage? {
         try? database.viewContext.message(id: id)?.asModel()
     }
+
+    /// Loads a thread model with a matching `parentMessageId` from the **local data store**.
+    ///
+    /// If the thread doesn't exist locally, it's recommended to fetch it with `messageController.loadThread()`.
+    ///
+    /// **Warning**: Should be called on the `main` thread only.
```

Comment:
Not in the scope of this PR, since this is everywhere. But we should probably do better here, maybe check the thread we're on?

Reply:
Yes we can probably add an assertion or warning